transformer_engine/pytorch/attention.py

```diff
@@ -1329,16 +1388,17 @@ def attn_forward_func_with_cp(
         ), f"QKV format of {qkv_format} is not supported with context parallelism!"
     assert(qkv_format != "sbhd" or use_fused_attention
         ), "FlashAttention does not support sbhd format!"
-    assert(not(qkv_format == "thd" and use_fused_attention)
-        ), "FusedAttention does not support thd format!"
-    assert (attn_mask_type in ["causal", "no_mask"]
+    assert (qkv_format != 'thd' or \
+            not use_fused_attention or \
+            attn_mask_type in ["padding", "padding_causal"]
         ), f"Mask type of {attn_mask_type} is not supported with context parallelism!"
```

Comment:
Maybe `f"""Context parallelism is not supported for {attn_mask_type} mask type and {qkv_format} format with {"FusedAttention" if use_fused_attention else "FlashAttention"}!"""`?

Reply:
Done